db/view/view.cc

```diff
@@ -449,15 +452,53 @@ row_marker view_updates::compute_row_marker(const clustering_row& base_row) cons
         return cell.is_live_and_has_ttl() ? row_marker(cell.timestamp(), cell.ttl(), cell.expiry()) : row_marker(cell.timestamp());
     }
 
-    return marker;
+    return base_row.marker();
 }
 
-deletable_row& view_updates::get_view_row(const partition_key& base_key, const clustering_row& update) {
-    std::vector<bytes> linearized_values;
-    auto get_value = boost::adaptors::transformed([&, this] (const column_definition& cdef) -> managed_bytes_view {
+std::vector<view_updates::view_row_entry>
+view_updates::get_view_rows(const partition_key& base_key, const clustering_row& update, const std::optional<clustering_row>& existing) {
+    // deque doesn't invalidate references at emplace_back.
+    std::deque<bytes> linearized_values;
+
+    struct bytes_view_with_action {
+        managed_bytes_view _view;
+        bytes_with_action::action _action = bytes_with_action::no_action{};
+        bytes_view_with_action(managed_bytes_view view, bytes_with_action::action action)
+            : _view(view)
+            , _action(action)
+        {}
+        bytes_view_with_action(managed_bytes_view view)
+            : _view(view)
+        {}
+        bytes_view_with_action(bytes_with_action&& bwa, std::deque<bytes>& linearized_values)
+            : bytes_view_with_action(managed_bytes_view(linearized_values.emplace_back(std::move(bwa._view))), bwa._action)
+        {}
+        static managed_bytes_view get_view(const bytes_view_with_action& bvwa) {
```

Comment:
this helper does not seem to be used, since you use public fields anyway

Reply:
it is used later, for instance:
```
const partition_key pkey = partition_key::from_range(boost::adaptors::transform(pk, bytes_view_with_action::get_view));
```